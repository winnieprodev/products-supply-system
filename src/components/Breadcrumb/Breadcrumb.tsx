import { ArrowRightIcon } from "components";
import React from "react";
import { Empty } from "utils/types";

import * as Styled from "./breadcrumb.styled";

const Breadcrumb = ({ children }: React.PropsWithChildren<Empty>) => (
  <Styled.Wrapper>{children}</Styled.Wrapper>
);

const Item = ({ children }: React.PropsWithChildren<Empty>) => (
  <Styled.ItemSection>
    {children}
    <Styled.Icon>
      <ArrowRightIcon />
    </Styled.Icon>
  </Styled.ItemSection>
);

Breadcrumb.Item = Item;

export default Breadcrumb;
