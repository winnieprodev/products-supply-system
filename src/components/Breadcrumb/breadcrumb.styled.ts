import styled from "styled-components";
import { primaryColor } from "styles/colors";

export const Wrapper = styled.div`
  margin-top: 20px;
  padding: 16px;
  display: flex;
`;

export const ItemSection = styled.div`
  display: flex;
  color: ${primaryColor};
  font-size: 15px;
  font-weight: 600;
  text-decoration: underline;
`;

export const Icon = styled.div`
  font-size: 15px;
  padding-left: 5px;
  padding-right: 5px;
  color: ${primaryColor};
`;
