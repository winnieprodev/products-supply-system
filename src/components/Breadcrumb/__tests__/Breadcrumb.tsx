import { render } from "@testing-library/react";

import Breadcrumb from "../Breadcrumb";

describe("Breadcrumb", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot for basic use", () => {
      const { container } = render(
        <Breadcrumb>
          <Breadcrumb.Item>Test1</Breadcrumb.Item>
          <Breadcrumb.Item>Test2</Breadcrumb.Item>
          <Breadcrumb.Item>Test3</Breadcrumb.Item>
          Test4
        </Breadcrumb>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
