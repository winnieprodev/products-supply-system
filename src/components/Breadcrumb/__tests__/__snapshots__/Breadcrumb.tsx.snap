// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb Snapshot tests it matches the snapshot for basic use 1`] = `
.c0 {
  margin-top: 20px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #F25C05;
  font-size: 15px;
  font-weight: 600;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 {
  font-size: 15px;
  padding-left: 5px;
  padding-right: 5px;
  color: #F25C05;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    Test1
    <div
      class="c2"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-angle-right fa-w-8 "
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    class="c1"
  >
    Test2
    <div
      class="c2"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-angle-right fa-w-8 "
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    class="c1"
  >
    Test3
    <div
      class="c2"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-angle-right fa-w-8 "
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  Test4
</div>
`;
